modules/react/layout/lib/utils/grid.ts

```diff
@@ -79,14 +107,22 @@ const gridProps = {
  * );
  *
  */
-export function grid<P extends GridStyleProps>(props: P) {
-  const styles = {};
+export function grid<P extends SpaceStyleProps>(props: P) {
+  let styles = {};
   for (const key in props) {
-    if (key in gridProps) {
-      const attr = gridProps[key as keyof GridStyleProps];
-      const value = props[key];
-      // @ts-ignore TS doesn't like adding a potentially unknown key to an object, but because we own this object, it's fine.
-      styles[attr] = value;
+    if (props.hasOwnProperty(key)) {
+      if (key in gridProps) {
+        const attr = gridProps[key as keyof GridStyleProps];
+        const value = props[key];
+        // @ts-ignore TS doesn't like adding a potentially unknown key to an object, but because we own this object, it's fine.
+        styles[attr] = value;
```

Comment:
Tiny nit: Let's add a `continue` here to tell the loop to skip the next `if` check.
```suggestion
        styles[attr] = value;
        continue;
```

Reply:
We do the same thing in our space fn [here](https://github.com/Workday/canvas-kit/blob/master/modules/react/layout/lib/utils/space.ts#L195).

Reply:
Done and updated